superset/db_engine_specs/ocient.py

```diff
@@ -215,32 +227,62 @@ class OcientEngineSpec(BaseEngineSpec):
 
     @classmethod
     def get_table_names(
-        cls, database: "Database", inspector: Inspector, schema: Optional[str]
+        cls, database: Database, inspector: Inspector, schema: Optional[str]
     ) -> List[str]:
         return sorted(inspector.get_table_names(schema))
 
     @classmethod
     def fetch_data(cls, cursor, lim=None):
         rows = super(OcientEngineSpec, cls).fetch_data(cursor)
+                
+        if len(rows) > 0 and type(rows[0]).__name__ == rows:
+            # Peek at the schema to determine which column values, if any,
+            # require sanitization.
+            columns_to_sanitize: List[PlacedSanitizeFunc] = _find_columns_to_sanitize(cursor)
 
-        if (len(rows)) == 0:
-            # No rows were produced
-            return rows
+            if columns_to_sanitize:
+                # At least 1 column has to be sanitized.
+                for row in rows:
+                    for info in columns_to_sanitize:
+                        # Modify the element in-place.
+                        v = row[info.column_index]
+                        row[info.column_index] = info.sanitize_func(v)
 
-        if type(rows[0]).__name__ != 'Row':
-            # TODO what else is returned here?
-            return rows
+        # # We are done with this cursor so we can safely remove it from the cache
+        OcientEngineSpec.query_id_mapping_lock.acquire()
+        del OcientEngineSpec.query_id_mapping[getattr(cursor, 'superset_query_id')]
+        OcientEngineSpec.query_id_mapping_lock.release()
```

Comment:
nit: a more pythonic and safe way to do this is with try-finally semantics. Python locks implement `__enter()__` and `__exit__()` (or w/e those methods are called) to provide the `with` statement. It's just syntactic sugar.
```suggestion
        with OcientEngineSpec.query_id_mapping_lock:
                del OcientEngineSpec.query_id_mapping[getattr(cursor, 'superset_query_id')]
```

Reply:
fixed